crates/requirements-txt/src/lib.rs

```diff
@@ -1279,4 +1325,34 @@ mod test {
             Some(("../editable[", "[dev]"))
         );
     }
+
+    #[test]
+    fn parser_error_line_and_column() -> Result<()> {
+        let temp_dir = assert_fs::TempDir::new()?;
+        let requirements_txt = temp_dir.child("requirements.txt");
+        requirements_txt.write_str(indoc! {"
+            numpy>=1,<2
+              --borken
+            tqdm
+        "})?;
+
+        let error = RequirementsTxt::parse(requirements_txt.path(), temp_dir.path()).unwrap_err();
+        let errors = anyhow::Error::new(error).chain().join("\n");
+
+        let requirement_txt =
+            regex::escape(&requirements_txt.path().simplified_display().to_string());
+        let filters = vec![
+            (requirement_txt.as_str(), "<REQUIREMENTS_TXT>"),
+            (r"\\", "/"),
+        ];
+        insta::with_settings!({
+            filters => filters
+        }, {
+            insta::assert_display_snapshot!(errors, @r###"
+            Unexpected '-', expected '-c', '-e', '-r' or the start of a requirement in `<REQUIREMENTS_TXT>` at position 2:3
+            "###);
+        });
+
+        Ok(())
+    }
```

Comment:
Could you add a couple of tests that include non-ASCII codepoints. More concretely, one test with a non-ASCII codepoint like, say, `💩`. And then another test with grapheme cluster made up of more than one codepoint like `à̖` (that's `a\u{0300}\u{0316}` as a string literal in Rust).

Reply:
Done in 223fd99ac1b44e0557227d1c461a20bf1a9d85f2